src/averageAges.js

```diff
@@ -55,7 +61,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenWithMothersListed = people
+    .filter(child => people.some(mother => mother.name === child.mother)
+      && (!onlyWithSon || child.sex === 'm'))
+    .map(child => ({
```

Comment:
Please take a look on previous mentor's comment. I suggest to move your filtered mothers into certain variable